### PR TITLE
Redundant type declaration linter rule

### PR DESCRIPTION
**Pull request by ivojawer**
Linter rule to help preventing the error declared in https://github.com/pharo-project/pharo-vm/pull/603

### CODE REVIEW

File: smalltalksrc/VMMakerTests/VMLiterRulesTest.class.st [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+		       'testMethod <var:#aVar declareC: #aDeclaration>'.
+	self assert:
+		(ReSlangRedundantTypeDeclarationRule new basicCheck:
+			 ast pragmas first).
+
+	ast := self class compiler parse:
+		       'testMethod <var:#aVar declareC: #aDeclaration> | aVar |'.
+	self deny:
+		(ReSlangRedundantTypeDeclarationRule new basicCheck:
+			 ast pragmas first)
```
Comment (PalumboN) [inline smalltalksrc/VMMakerTests/VMLiterRulesTest.class.st:42]: Should be nice have one for parameters also.

And could be different tests :)

**Review by PalumboN — APPROVED** [review]
Cooooollllllll

**Review by guillep — APPROVED** [review]
LGTM, does this run by default without configuring anything? :O

**Comment by PalumboN** [comment]
Yes, by default looks for all the subclasses. Maybe you need to reset the Manager cache:

```st
ReRuleManager reset
```

**Comment by guillep** [comment]
Merci gohan

**Pull request merged**